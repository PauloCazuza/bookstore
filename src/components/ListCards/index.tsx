import { Box, Center, FlatList, HStack, Text, VStack } from "native-base";
import { memo } from "react";
import { IBook } from "../../interfaces/Book/IBook";
import { IFormatList } from "../../interfaces/FormatData";
import Card from "../Card";
import Spinner from "../Spinner";

type ListCardProps = {
    listItems: IFormatList[];
    actionEndReached?: VoidFunction;
    modeView?: "unic" | "double";
    loading?: boolean;
}

function ListCards({ listItems, actionEndReached, modeView = "unic", loading }: ListCardProps) {

    function modeUnic(left: IBook, right: IBook) {
        return (
            <VStack>
                <Box flex={1} py={1} >
                    {left && <Card book={left} />}

                </Box>
                <Box flex={1} py={1} >
                    {right && < Card book={right} />}
                </Box>
            </VStack>
        )
    }

    function modeDouble(left: IBook, right: IBook) {
        return (
            <HStack py={1}>
                <Box flex={1} px={1} >
                    {left && <Card book={left} />}

                </Box>
                <Box flex={1} px={1} >
                    {right && < Card book={right} />}
                </Box>
            </HStack>
        )
    }

    return (
        <FlatList
            data={listItems}
            marginTop={2}
            width="100%"
            initialNumToRender={4}
            removeClippedSubviews
            keyExtractor={item => item.left?.id + item.right?.id}
            flexGrow={1}
            renderItem={({
                item: { left, right }
            }) => {
                return modeView === "unic" ? modeUnic(left, right) : modeDouble(left, right)
            }}
            // ListEmptyComponent={() => <Center> <Text fontSize={30}> Sem Resultados. </Text> </Center>}
            ListFooterComponent={listItems.length > 10 && <Spinner />}
            onEndReached={actionEndReached}
        />
    )
}

export default ListCards;