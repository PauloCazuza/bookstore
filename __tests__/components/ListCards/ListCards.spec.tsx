import { NativeBaseProvider } from "native-base";
import renderer from "react-test-renderer";
import ListCards from "../../../src/components/ListCards/index";

it('renders correctly ListCards', () => {
    const tree = renderer.create(<NativeBaseProvider><ListCards listItems={[]} /></NativeBaseProvider>).toJSON();
    expect(tree).toMatchSnapshot();
});