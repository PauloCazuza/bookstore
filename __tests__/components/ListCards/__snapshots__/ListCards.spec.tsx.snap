// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly ListCards 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    Array [
      Object {
        "flex": 1,
      },
      undefined,
    ]
  }
/>
`;
